code/xcvm/cosmwasm/contracts/gateway/src/contract/ibc/ics20.rs

```diff
@@ -0,0 +1,141 @@
+//! Helps connecting identifiers into networks.
+//! Allows to map asset identifiers, contracts, networks, channels, denominations from, to and on
+//! each chain via contract storage, precompiles, host extensions.
+//! handles PFM and IBC wasm hooks
+use cosmwasm_std::{
+	to_binary, wasm_execute, Addr, Binary, Coin, DepsMut, Env, MessageInfo, Response, Storage,
+	SubMsg,
+};
+use cw_xc_interpreter::msg;
+use ibc_rs_scale::core::ics24_host::identifier::ChannelId;
+use xc_core::{
+	gateway::{Asset, ExecuteMsg, ExecuteProgramMsg, GatewayId},
+	ibc::{to_cw_message, IbcRoute, Ics20MessageHook, WasmMemo},
+	proto::{decode_packet, Encodable},
+	shared::{DefaultXCVMProgram, XcPacket},
+	AssetId, CallOrigin, Funds,
+};
+
+use crate::{
+	auth,
+	contract::EXEC_PROGRAM_REPLY_ID,
+	error::{ContractError, ContractResult},
+	events::make_event,
+	state,
+	state::{NetworkItem, OtherNetworkItem},
+};
+
+pub(crate) fn handle_bridge_forward(
```

Comment:
what does handle bridge forward means?

Reply:
but is not sent too, because there is deeper method which sends